### PR TITLE
Remove magic numbers for keycodes and centralize keycodes.

### PR DESCRIPTION
**Pull request by BugsNash**

### CODE REVIEW

File: js/controllers/hotkeys.js [hunk 1]
```diff
@@ -37,77 +45,77 @@ function HotkeysController(windowController, tabs, editor, settings, analytics)
 HotkeysController.prototype.onKeydown_ = function(e) {
   if (e.ctrlKey || e.metaKey) {
     switch (e.keyCode) {
```
Comment (E-T-) [inline js/controllers/hotkeys.js]: Way easier than this! Let's stop using `e.keycode` and start using `e.key`!

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

also note that the [compatibility table](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#Browser_compatibility) says that it should be fine for our versions of Chrome so I think this is a much better way to handle this :)

File: js/controllers/hotkeys.js [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+  SPACE: 32,
+  ZERO: 48,
+  NUMPAD_ZERO: 96,
+  NUMPAD_PLUS: 107,
+  NUMPAD_MINUS: 109,
+  PLUS: 187,
+  MINUS: 189
+};
+// Populate Keycode enum A - Z
+for (var i = 65; i <= 90; i++) {
```
Comment (E-T-) [inline js/controllers/hotkeys.js]: Note for future: prefer let over var :)

File: js/controllers/hotkeys.js [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+  TAB: 9,
+  ENTER: 13,
+  ESCAPE: 27,
+  SPACE: 32,
+  ZERO: 48,
+  NUMPAD_ZERO: 96,
+  NUMPAD_PLUS: 107,
+  NUMPAD_MINUS: 109,
+  PLUS: 187,
+  MINUS: 189
```
Comment (E-T-) [inline js/controllers/hotkeys.js]: Note for future: add a trailing comma to the end of object literals like this :)

File: js/controllers/hotkeys.js [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+};
+// Populate Keycode enum A - Z
+for (var i = 65; i <= 90; i++) {
+  Keycode[String.fromCharCode(i).toUpperCase()] = i
```
Comment (E-T-) [inline js/controllers/hotkeys.js]: Note for future: add semicolons :)

File: js/controllers/hotkeys.js (hunk 3 above)
Comment (BugsNash) [reply js/controllers/hotkeys.js]: Ack.

File: js/controllers/hotkeys.js (hunk 2 above)
Comment (BugsNash) [reply js/controllers/hotkeys.js]: Ah yeah I just moved this code so I didn't notice. Ack.

File: js/controllers/hotkeys.js (hunk 4 above)
Comment (BugsNash) [reply js/controllers/hotkeys.js]: Ack.

File: js/controllers/hotkeys.js (hunk 1 above)
Comment (BugsNash) [reply js/controllers/hotkeys.js]: Done, thanks for pointing that out!

File: js/controllers/search.js [hunk 5]
```diff
@@ -67,13 +67,13 @@ SearchController.prototype.onChange_ = function() {
 };
 
 SearchController.prototype.onKeydown_ = function(e) {
-  switch (e.keyCode) {
-    case 13:
+  switch (e.key) {
+    case "Enter":
```
Comment (E-T-) [inline js/controllers/search.js]: Throughout this patch:

prefer `'` over `"` for JavaScript. prefer `"` when doing HTML.

File: js/controllers/hotkeys.js [hunk 6]
```diff
@@ -36,78 +21,78 @@ function HotkeysController(windowController, tabs, editor, settings, analytics)
...
         return false;
 
-      case this.KEY.MINUS:
-      case this.KEY.NUMPAD_MINUS:
+      case "-":
+      case "_":
         var fontSize = this.settings_.get('fontsize');
         this.settings_.set('fontsize', fontSize * this.ZOOM_OUT_FACTOR);
         return false;
     }
```
Comment (E-T-) [inline js/controllers/hotkeys.js]: add a default case here:

https://google.github.io/styleguide/jsguide.html#features-switch-statements

File: js/controllers/search.js (hunk 5 above)
Comment (BugsNash) [reply js/controllers/search.js]: Done here and in other file, thanks for catching that I need to get better at these :\

File: js/controllers/hotkeys.js (hunk 6 above)
Comment (BugsNash) [reply js/controllers/hotkeys.js]: Done

**Pull request merged**